backend/internal/userschema/constants/errorconstants.go

```diff
@@ -0,0 +1,95 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package constants defines error constants for user schema management operations.
+package constants
+
+import (
+	"errors"
+
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+)
+
+// Client errors for user schema management operations.
+var (
+	// ErrorInvalidRequestFormat is the error returned when the request format is invalid.
+	ErrorInvalidRequestFormat = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1001",
+		Error:            "Invalid request format",
+		ErrorDescription: "The request body is malformed or contains invalid data",
+	}
+	// ErrorUserSchemaNotFound is the error returned when a user schema is not found.
+	ErrorUserSchemaNotFound = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1002",
+		Error:            "User schema not found",
+		ErrorDescription: "The user schema with the specified id does not exist",
+	}
+	// ErrorUserSchemaNameConflict is the error returned when user schema name already exists.
+	ErrorUserSchemaNameConflict = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1003",
+		Error:            "User schema name conflict",
+		ErrorDescription: "A user schema with the same name already exists",
+	}
+	// ErrorInvalidUserSchemaRequest is the error returned when user schema request is invalid.
+	ErrorInvalidUserSchemaRequest = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1004",
+		Error:            "Invalid user schema request",
+		ErrorDescription: "The user schema request contains invalid or missing required fields",
+	}
+	// ErrorInvalidLimit is the error returned when limit parameter is invalid.
+	ErrorInvalidLimit = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1005",
+		Error:            "Invalid pagination parameter",
+		ErrorDescription: "The limit parameter must be a positive integer",
+	}
+	// ErrorInvalidOffset is the error returned when offset parameter is invalid.
+	ErrorInvalidOffset = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "USRS-1006",
+		Error:            "Invalid pagination parameter",
+		ErrorDescription: "The offset parameter must be a non-negative integer",
+	}
+)
+
+// Server errors for user schema management operations.
+var (
+	// ErrorInternalServerError is the error returned when an internal server error occurs.
+	ErrorInternalServerError = serviceerror.ServiceError{
+		Type:             serviceerror.ServerErrorType,
+		Code:             "USRS-5000",
```

Comment:
Shall we start error codes with 5001?

Reply:
Since we don't share any details with the internal server error, it will have only one error code for internal server error scenario..

With that in mind, it would be better to have 5000 as the error code, rather than the 5001.

This has been the pattern that followed in other places as well.

Reply:
We will have to fix some places that starts with 5001 then